src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,207 @@
+ * creating tasks, adding or removing tasks from the list, marking
+ * tasks as completed, and generating a print message of the list.
+ */
+public class TaskList {
+
+    /** The list containing Tasks objects. */
+    private final List<Task> list;
+
+    /**
+     * Constructor for TaskList. This constructor is usually
+     * used when loading a list of tasks from the txt file.
+     * @param list the list of tasks.
+     */
+    public TaskList(List<Task> list) {
+        this.list = list;
+    }
+
+    /**
+     * Constructor for TaskList to create an empty list.
+     */
+    public TaskList() {
+        this.list = new ArrayList<>();
+    }
+
+    /**
+     * Returns the size of the list.
+     * @return the size of the list.
+     */
+    public int size() {
+        return list.size();
+    }
+
+    /**
+     * Returns the task list object.
+     * @return the task list object.
+     */
+    public List<Task> getList() {
+        return list;
+    }
+
+    /**
+     * The static factory method to create the subtypes of Tasks:
+     *
+     *     - Todo: A simple task with a description only
+     *     - Event: A task requiring attendance at a date and time
+     *     - Deadline: A task requiring completion by a date and time
+     *
+     * Creation of these tasks are based on the type input. After which,
+     * the description of the task will be parsed and handled individually
+     * by the appropriate subclasses.
+     * @param type the subtype of the Task to be created.
+     * @param description the raw description of the task.
+     * @return the created Task object.
+     * @throws DukeException when creation fails usually due to improper format description.
+     */
+    public static Task createTask(String type, String description) throws DukeException {
+        Task task = new Todo("");
+        String[] split;
+        String desc;
+        description = description.trim();
+        if (description.isEmpty()) {
+            throw new DukeException("Description cannot be empty!");
+        }
+        switch (type) {
+        case "todo":
+            task = new Todo(description);
+            break;
+        case "deadline":
+            try {
+                split = description.split(" /by ");
+                desc = split[0];
+                String by = split[1];
+                task = new Deadline(desc, by);
+            } catch (IndexOutOfBoundsException ioobe) {
+                throw new DukeException("Invalid " + type + " description!");
+            }
+            break;
+        case "event":
+            try {
+                split = description.split(" /at ");
+                desc = split[0];
+                String at = split[1];
+                task = new Event(desc, at);
+            } catch (IndexOutOfBoundsException ioobe) {
+                throw new DukeException("Invalid " + type + " description!");
+            }
+            break;
+        }
```

Comment:
You might want to include a default statement here? I believe this will ensure the 'type' you pass into this function has no spelling errors for easier debugging.